### PR TITLE
Use libipld-json to replace ipld/src/json.rs 

### LINKED ISSUE #1700: Use `libipld-json` to replace `ipld/src/json.rs`
**Issue summary**

The [libipld-json](https://crates.io/crates/libipld-json) crate is nearly identical to our own JSON serializer. Replacing our code will require some refactoring, though. If the required changes are too large, update this issue with a summary.

**Other information and links**

### PR DESCRIPTION
**Pull request by tyshko5**
**Summary of changes**
Changes introduced in this pull request:
- 

**Reference issue to close (if applicable)**

Closes #1700

**Other information and links**

### CODE REVIEW

**Comment by lemmih** [comment]
Hmm, it seems that the tests in `walk_tests.rs` were accidentally disabled (hidden by the `submodule_tests` feature flag). It would be great if you could re-enable them.

**Comment by tyshko5** [comment]
> Hmm, it seems that the tests in `walk_tests.rs` were accidentally disabled (hidden by the `submodule_tests` feature flag). It would be great if you could re-enable them.

These tests seem to be broken some time ago

**Comment by lemmih** [comment]
> > Hmm, it seems that the tests in `walk_tests.rs` were accidentally disabled (hidden by the `submodule_tests` feature flag). It would be great if you could re-enable them.
> 
> These tests seem to be broken some time ago

Think you can fix them or should we open a new issue for that?

**Comment by tyshko5** [comment]
> > > Hmm, it seems that the tests in `walk_tests.rs` were accidentally disabled (hidden by the `submodule_tests` feature flag). It would be great if you could re-enable them.
> > 
> > 
> > These tests seem to be broken some time ago
> 
> Think you can fix them or should we open a new issue for that?

I can try. I think a new issue, as the problems there are more connected with changing of Blockstore implementation for MemoryDB, rather than ipld json

File: node/rpc-api/src/data_types.rs [hunk 1]
```diff
@@ -1,24 +1,20 @@
...
-use jsonrpc_v2::{MapRouter as JsonRpcMapRouter, Server as JsonRpcServer};
-use serde::{Deserialize, Serialize};
-
-use actor::market::{DealProposal, DealState};
 use beacon::{json::BeaconEntryJson, Beacon, BeaconSchedule};
 use chain::{headchange_json::SubscriptionHeadChange, ChainStore};
 use chain_sync::{BadBlockCache, SyncState};
 use cid::Cid;
+use fil_types::SectorSize;
 use fil_types::{json::SectorInfoJson, sector::post::json::PoStProofJson};
```
Comment (lemmih) [inline node/rpc-api/src/data_types.rs:13]:
```suggestion
use fil_types::{SectorSize, json::SectorInfoJson, sector::post::json::PoStProofJson};
```

**Review by lemmih — APPROVED** [review]
I'm surprised this was so easy. But I guess the tests don't lie.

**Review by LesnyRumcajs — APPROVED** [review]

**Pull request merged**